pyanaconda/modules/common/structures/partitioning.py

```diff
@@ -343,3 +343,233 @@ def format_options(self) -> Str:
     @format_options.setter
     def format_options(self, options: Str):
         self._format_options = options
+
+
+class DeviceFactoryRequest(DBusData):
+    """Device factory request data."""
+
+    def __init__(self):
+        self._device_type = 0
+        self._device_spec = ""
+        self._disks = []
+
+        self._mount_point = ""
+        self._format_type = ""
+        self._label = ""
+        self._luks_version = ""
+
+        self._device_name = ""
+        self._device_size = 0
+        self._device_raid_level = ""
+        self._device_encrypted = False
+
+        self._container_name = ""
+        self._container_size_policy = 0
+        self._container_raid_level = ""
+        self._container_encrypted = False
+
+    @property
+    def device_type(self) -> Int:
+        """Type of the device.
+
+        Supported values:
+            0  LVM
+            1  RAID
+            2  Standard Partition
+            3  Btrfs
+            4  Disk
+            5  LVM Thin Provisioning
+
+        :return: a number of the type
+        """
+        return self._device_type
```

Comment:
This value can be directly propagated to Blivet, so it is simpler to work with. Also enums are not supported by pydbus or GLib, so it would require some workarounds. I will keep it in mind for the future improvements.

Reply:
We are using Enums with DBus even now by sending value and that's fine. But yeah if it makes your life easier to work with blivet than it's fine to have it this way.

Reply:
I would like to eventually figure out how to improve the support for Enums, but I don't think it is possible with pydbus.

Reply:
ok